aws-cert-converter.py

```diff
@@ -0,0 +1,132 @@
+import argparse
+from os import walk
+
+# If modifying the template below,
+# make sure to double escape newlines you want in the C-string (ie: '\\n')
+# and to double-up brackets ('{' and '}' become '{{' and '}}') or else
+# formatting of the template string will fail
+header_template = (
+    '// AWS Certificates\n'
+    '\n'
+    '#ifndef AWS_CREDENTIALS_H\n'
+    '#define AWS_CREDENTIALS_H\n'
+    '\n'
+    'namespace aws {{\n'
+    'namespace credentials {{\n'
+    '/*\n'
+    ' * PEM-encoded root CA certificate\n'
+    ' *\n'
+    ' * Must include the PEM header and footer,\n'
+    ' * and every line of the body needs to be quoted and end with \\n:\n'
+    ' * "-----BEGIN CERTIFICATE-----\\n"\n'
+    ' * "...base64 data...\\n"\n'
+    ' * "-----END CERTIFICATE-----";\n'
+    ' */\n'
+    'const char rootCA[] = \n'
+    '{0};\n'
+    '\n'
+    '/*\n'
+    ' * PEM-encoded client certificate\n'
+    ' *\n'
+    ' * Must include the PEM header and footer,\n'
+    ' * and every line of the body needs to be quoted and end with \\n:\n'
+    ' * "-----BEGIN CERTIFICATE-----\\n"\n'
+    ' * "...base64 data...\\n"\n'
+    ' * "-----END CERTIFICATE-----";\n'
+    ' */\n'
+    'const char clientCrt[] = \n'
+    '{1};\n'
+    '\n'
+    '/*\n'
+    ' * PEM-encoded client private key.\n'
+    ' *\n'
+    ' * Must include the PEM header and footer,\n'
+    ' * and every line of the body needs to be quoted and end with \\n:\n'
+    ' * "-----BEGIN RSA PRIVATE KEY-----\\n"\n'
+    ' * "...base64 data...\\n"\n'
+    ' * "-----END RSA PRIVATE KEY-----";\n'
+    ' */\n'
+    'const char clientKey[] = \n'
+    '{2};\n'
+    '\n'
+    '}}\n'
+    '}}\n'
+    '\n'
+    '#endif\n'
+)
```

Comment:
Python's multiline strings make life a lot easier:
```suggestion
header_template = (
    """
// AWS Certificates

#ifndef AWS_CREDENTIALS_H
#define AWS_CREDENTIALS_H

namespace aws {{
namespace credentials {{
/*
    * PEM-encoded root CA certificate
    *
    * Must include the PEM header and footer,
    * and every line of the body needs to be quoted and end with \\n:
    * "-----BEGIN CERTIFICATE-----\\n"
    * "...base64 data...\\n"
    * "-----END CERTIFICATE-----";
    */
const char rootCA[] = 
{0};

/*
    * PEM-encoded client certificate
    *
    * Must include the PEM header and footer,
    * and every line of the body needs to be quoted and end with \\n:
    * "-----BEGIN CERTIFICATE-----\\n"
    * "...base64 data...\\n"
    * "-----END CERTIFICATE-----";
    */
const char clientCrt[] = 
{1};

/*
    * PEM-encoded client private key.
    *
    * Must include the PEM header and footer,
    * and every line of the body needs to be quoted and end with \\n:
    * "-----BEGIN RSA PRIVATE KEY-----\\n"
    * "...base64 data...\\n"
    * "-----END RSA PRIVATE KEY-----";
    */
const char clientKey[] = 
{2};

}}
}}

#endif
    """
)
```

Reply:
Also, we may not need comments/docs in the template since the generated file already contains the keys? And the format (e.g. `\n`) will be obvious to users if they want to replace any keys.